renderer/components/ConfigBox.tsx

```diff
@@ -1,77 +1,98 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import {
-  Modal,
-  Button,
-  FormGroup,
-  Form,
-  FormControl,
-  ControlLabel,
-} from 'react-bootstrap';
+import { Modal, Button, FormGroup, Form, FormControl, ControlLabel } from 'react-bootstrap';
 import { remote, ipcRenderer } from 'electron';
 import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
 import _ from 'lodash';
 import { getValidationState, logging } from '../utils/utils';
 import { updateFieldControl } from '../actions/FieldActions';
 import { ipChange } from '../actions/InfoActions';
 
-
 const storage = remote.require('electron-json-storage');
 
-class ConfigBoxComponent extends React.Component {
-  constructor(props) {
+interface Config {
+  stationNumber: number;
+  bridgeAddress: string;
+}
+
+interface StateProps {
+  stationNumber: number;
+  ipAddress: string;
+}
+
+interface DispatchProps {
+  onIPChange: (ipAddress: string) => void;
+  onFCUpdate: (config: Config) => void;
+}
+
+interface OwnProps {
+  fcAddress: string;
+  shouldShow: boolean;
+  hide: () => void;
+}
+
+type Props = StateProps & DispatchProps & OwnProps;
```

Comment:
@lancejackson @HaileyJang Note the separation of different props, props that are derived from `ApplicationState` should be in the `StateProps` interface. `DispatchProps` are actions that then get dispatched to the reducers. `OwnProps` are the ones that are passed to the component from another component.